### PR TITLE
Extends support of Oneview_uplink_set for API300

### PR DESCRIPTION
**Pull request by marikrg**
### Description
Use Oneview_uplink_set with API 300 on the Synergy and C7000 hardware variants

### Issues Resolved
https://github.com/HewlettPackard/oneview-puppet/issues/32

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Comment by marikrg** [comment]
I'm fixing the broken build.

**Review by fgbulsoni — CHANGES_REQUESTED** [review]
line 42:     
``@property_hash[:data] = @data``
should be
``@property_hash[:data] = uplink_set.data``

Also, found and destroy methods are being inherited and could be dropped.

**Comment by marikrg** [comment]
> I'm fixing the broken build.

Done.

**Comment by marikrg** [comment]
> line 42:
> @property_hash[:data] = @data
> should be
> @property_hash[:data] = uplink_set.data
> 
> Also, found and destroy methods are being inherited and could be dropped.

Done.

**Comment by marikrg** [comment]
I just noticed that the update is not working properly for Uplink Sets. I will fix it.

**Comment by marikrg** [comment]
> I just noticed that the update is not working properly for Uplink Sets. I will fix it.

As we decided, we are going to fix the update later, since this resource needs a different approach (see the bug opened for oneview-ansible at: https://github.com/HewlettPackard/oneview-ansible/issues/13)

**Review by fgbulsoni — APPROVED** [review]

File: spec/unit/provider/oneview_uplink_set_c7000_spec.rb [hunk 1]
```diff
@@ -18,18 +18,17 @@
 require_relative '../../support/fake_response'
 require_relative '../../shared_context'
 
-provider_class = Puppet::Type.type(:oneview_uplink_set).provider(:oneview_uplink_set)
+provider_class = Puppet::Type.type(:oneview_uplink_set).provider(:c7000)
 resourcetype = OneviewSDK::UplinkSet
```
Comment (fgbulsoni) [inline spec/unit/provider/oneview_uplink_set_c7000_spec.rb]: Should add something along the lines of the following code to account for different APIs
```ruby
api_version = login[:api_version] || 200
resource_name = 'UplinkSet'
resourcetype = if api_version == 200
                 Object.const_get("OneviewSDK::API#{api_version}::#{resource_name}")
               else
                 Object.const_get("OneviewSDK::API#{api_version}::C7000::#{resource_name}")
               end
```

File: spec/unit/provider/oneview_uplink_set_synergy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+require_relative '../../support/fake_response'
+require_relative '../../shared_context'
+
+provider_class = Puppet::Type.type(:oneview_uplink_set).provider(:synergy)
+resourcetype = OneviewSDK::UplinkSet
```
Comment (fgbulsoni) [inline spec/unit/provider/oneview_uplink_set_synergy_spec.rb]: Should add something along the lines of:
```ruby
api_version = login[:api_version] || 200
resource_name = 'UplinkSet'
resourcetype = Object.const_get("OneviewSDK::API#{api_version}::Synergy::#{resource_name}") unless api_version < 300
```
to account for different apis

File: spec/unit/provider/oneview_uplink_set_synergy_spec.rb [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+describe provider_class, unit: true do
```
Comment (fgbulsoni) [inline spec/unit/provider/oneview_uplink_set_synergy_spec.rb]: should add 
```ruby
if: api_version >= 300
```
to avoid this test running for API 200

**Review by fgbulsoni — CHANGES_REQUESTED** [review]

**Review by fgbulsoni — APPROVED** [review]

**Pull request merged**